src/elmer_compiler.erl

```diff
@@ -60,6 +60,10 @@ dump(erlsrc, Filename, Erl) ->
     Bin = erlang:iolist_to_binary(Src),
     {Filename, Bin};
 
+dump(binary, Filename, Erl) ->
```

Comment:
I think this is the only thing that should be merged then. Perhaps changing the atom to `beam` could be more explicit.
I'll fix the test to only verify that a compiled beam can be loaded and executed from erlang.